### PR TITLE
Updated crd names operator version to be compliant with operator hub

### PR DESCRIPTION
**Pull request by tmihalac**
Signed-off-by: Theodor Mihalache <tmihalac@redhat.com>

### CODE REVIEW

**Comment by ruromero** [comment]
I see, then I guess you have to address the rest of the warnings coming from the bundle validate and also update the workflow to use the `operatorframework` suite

```
↳ operator-sdk bundle validate ./bundle --select-optional suite=operatorframework
WARN[0000] Warning: Value : (mercury-operator.1.0.2) csv.metadata.Name mercury-operator.1.0.2 is not following the recommended naming convention: <operator-name>.v<semver> e.g. memcached-operator.v0.0.1 
WARN[0000] Warning: Value mercury-operator.1.0.2: unable to find the resource requests for the container: (mercury-operator). It is recommended to ensure the resource request for CPU and Memory. Be aware that for some clusters configurations it is required to specify requests or limits for those values. Otherwise, the system or quota may reject Pod creation. More info: https://master.sdk.operatorframework.io/docs/best-practices/managing-resources/ 
WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "kafkas.kafka.strimzi.io" has an empty description 
WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "kafkatopics.kafka.strimzi.io" has an empty description 
WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "kafkausers.kafka.strimzi.io" has an empty description 
WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "integrations.camel.apache.org" has an empty description 
```

**Comment by tmihalac** [comment]
> I see, then I guess you have to address the rest of the warnings coming from the bundle validate and also update the workflow to use the `operatorframework` suite
> 
> ```
> ↳ operator-sdk bundle validate ./bundle --select-optional suite=operatorframework
> WARN[0000] Warning: Value : (mercury-operator.1.0.2) csv.metadata.Name mercury-operator.1.0.2 is not following the recommended naming convention: <operator-name>.v<semver> e.g. memcached-operator.v0.0.1 
> WARN[0000] Warning: Value mercury-operator.1.0.2: unable to find the resource requests for the container: (mercury-operator). It is recommended to ensure the resource request for CPU and Memory. Be aware that for some clusters configurations it is required to specify requests or limits for those values. Otherwise, the system or quota may reject Pod creation. More info: https://master.sdk.operatorframework.io/docs/best-practices/managing-resources/ 
> WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "kafkas.kafka.strimzi.io" has an empty description 
> WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "kafkatopics.kafka.strimzi.io" has an empty description 
> WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "kafkausers.kafka.strimzi.io" has an empty description 
> WARN[0000] Warning: Value mercury-operator.1.0.2: required CRD "integrations.camel.apache.org" has an empty description 
> ```

image
I am not sure what values to specify for these two warnings

**Comment by ruromero** [comment]
For minKubeVersion I'd put 1.19 which matches with OCP 4.6 and in 1.19 some of the features we depend on were introduced. We can try to validate the installation on that version to confirm it works.

For the application resources/limits try to see what the operator is usually consuming and put something a bit over these values

**Comment by tmihalac** [comment]
> For minKubeVersion I'd put 1.19 which matches with OCP 4.6 and in 1.19 some of the features we depend on were introduced. We can try to validate the installation on that version to confirm it works.
> 
> For the application resources/limits try to see what the operator is usually consuming and put something a bit over these values

image
This is a screenshot of the operator pod with one sd ideal, how much cpu and memory will suffice in your opinion ?

**Comment by ruromero** [comment]
My suggestion is to add this to the mercury-operator.csv.yaml
```
spec:
  MinKubeVersion: 1.19.0
```

And this to the operator.yaml
```
        resources:
          limits:
            memory: "64Mi"
            cpu: "100m"
          requests:
            memory: "64Mi"
            cpu: "100m"
```

Don't forget to re-generate the CSV and the catalog

File: pom.xml [hunk 1]
```diff
@@ -98,7 +98,7 @@
         <!-- Container images -->
         <quarkus.container-image.registry>quay.io</quarkus.container-image.registry>
         <quarkus.container-image.group>ecosystem-appeng</quarkus.container-image.group>
-        <quarkus.container-image.tag>1.0.2</quarkus.container-image.tag>
+        <quarkus.container-image.tag>v1.0.2</quarkus.container-image.tag>
```
Comment (ruromero) [inline pom.xml:101]: If you change the container image tag, you have to update all the references to the image in the examples as well (e.g. party-routing-profile-example:1.0.2 -> party-routing-profile-example:v1.0.2)

File: deploy/olm-catalog/1.0.2/bundle/metadata/annotations.yaml [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: mercury-operator
+  operators.operatorframework.io.bundle.channels.v1: alpha
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.19.0+git
+  operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
+  operators.operatorframework.io.metrics.project_layout: unknown
+
+  # Annotations for testing.
+  operators.operatorframework.io.test.mediatype.v1: scorecard+v1
+  operators.operatorframework.io.test.config.v1: tests/scorecard/
```
Comment (ruromero) [inline deploy/olm-catalog/1.0.2/bundle/metadata/annotations.yaml:14]: I don't see this folder, see here for reference:
https://sdk.operatorframework.io/docs/testing-operators/scorecard/

File: deploy/olm-catalog/1.0.2/mercury-operator.csv.yaml [hunk 3]
```diff
@@ -131,6 +131,7 @@ spec:
     - name: GitHub Repository
       url: https://github.com/open-accelerators/mercury
   version: 1.0.2
```
Comment (ruromero) [inline deploy/olm-catalog/1.0.2/mercury-operator.csv.yaml:133]: I think it's better to rename all `1.0.2` to `v1.0.2` that affects the bundle and catalog images too

**Review by ruromero — CHANGES_REQUESTED** [review]

File: deploy/olm-catalog/1.0.2/bundle/metadata/annotations.yaml (hunk 2 above)
Comment (tmihalac) [reply deploy/olm-catalog/1.0.2/bundle/metadata/annotations.yaml:14]: I will remove those, it was auto generated

File: deploy/olm-catalog/1.0.2/mercury-operator.csv.yaml (hunk 3 above)
Comment (tmihalac) [reply deploy/olm-catalog/1.0.2/mercury-operator.csv.yaml:133]: I looked at a lot of other operators the version tag in all of them is without the v prefix, I guess that's the way it should be

**Comment by ruromero** [comment]
@tmihalac there should be only one commit on top of the main branch

File: deploy/olm-catalog/1.0.2/bundle/metadata/annotations.yaml [hunk 4]
```diff
@@ -1,7 +1,10 @@
 annotations:
-  operators.operatorframework.io.bundle.channel.default.v1: stable
-  operators.operatorframework.io.bundle.channels.v1: dev,stable
-  operators.operatorframework.io.bundle.manifests.v1: manifests/
+  # Core bundle annotations.
   operators.operatorframework.io.bundle.mediatype.v1: registry+v1
+  operators.operatorframework.io.bundle.manifests.v1: manifests/
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: mercury-operator
+  operators.operatorframework.io.bundle.channels.v1: alpha
```
Comment (ruromero) [inline deploy/olm-catalog/1.0.2/bundle/metadata/annotations.yaml:7]: the channels we're building are dev and stable

**Review by ruromero — CHANGES_REQUESTED** [review]

**Pull request merged**